lib/zabbix/zabbix.go

```diff
@@ -159,8 +171,44 @@ func (a *API) GetHostsInfo() (Results, error) {
 	}
 
 	var r struct {
-		Results `json:"result"`
-		Err     apiError `json:"error"`
+		HostResults `json:"result"`
+		Err         apiError `json:"error"`
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&r)
+	if err != nil {
+		return nil, fmt.Errorf("cannot decode response. error: %v", err)
+	}
+
+	//  Check if the response contains an error
+	if r.Err != (apiError{}) {
```

Comment:
If you make apiError a pointer in L175 above, coupled with the `omitempty` json annotation, you can use the familiar `r.Err != nil` here. But yea, this works also :)